File: app/src/main/java/com/shui520it/access/funcion/service/ShuiService.java
package com.shui520it.access.funcion.service;

import android.accessibilityservice.AccessibilityService;
import android.content.Intent;
import android.view.accessibility.AccessibilityEvent;

import com.shui520it.access.funcion.pool.ShuiRunnableImpl;
import com.shui520it.access.funcion.pool.ShuiThreadPool;

/**
 * @author shuimu{lwp}
 * @time 2019/8/14  11:59
 * @desc
 */
public class ShuiService extends AccessibilityService {

    @Override
    public void onCreate() {
        super.onCreate();
        System.out.println(">>>>>>>>>>服务被启动了 onCreate");
        ShuiThreadPool.get().init();
    }

    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        System.out.println(">>>>>>>>>>服务被启动了 onStartCommand");
        ShuiThreadPool.get().init();
        return super.onStartCommand(intent, flags, startId);
    }

    /**
     * 开启无障碍服务后，将会得到监听页面的所有事件。
     */
    @Override
    public void onAccessibilityEvent(AccessibilityEvent event) {
        int eventType = event.getEventType();
        if (AccessibilityEvent.TYPE_WINDOW_CONTENT_CHANGED == eventType) {
            ShuiThreadPool.get().execute(new ShuiRunnableImpl(this));
        }
    }

    @Override
    public void onInterrupt() {
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        System.out.println(">>>>>>>>>>服务被销毁了..");
        ShuiThreadPool.get().shutdown();
    }
}

File: app/src/main/java/com/shui520it/access/funcion/service/ShuiService.kt
package com.shui520it.access.funcion.service

import android.accessibilityservice.AccessibilityService
import android.content.Intent
import android.view.accessibility.AccessibilityEvent
import com.shui520it.access.funcion.pool.ShuiRunnableImpl
import com.shui520it.access.funcion.pool.ShuiThreadPool

/**
 * @author shuimu{lwp}
 * @time 2019/8/14  11:59
 * @desc
 */
class ShuiService : AccessibilityService() {
    override fun onCreate() {
        super.onCreate()
        ShuiThreadPool.runnable
        println(">>>>>>>>>>服务被启动了 onCreate")
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        println(">>>>>>>>>>服务被启动了 onStartCommand")
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onInterrupt() {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun onAccessibilityEvent(event: AccessibilityEvent?) {
        when (event?.eventType) {
            AccessibilityEvent.TYPE_WINDOW_CONTENT_CHANGED -> {
                //加个判断避免重复传教Runnable。
                if (ShuiThreadPool.runnable) {
                    ShuiThreadPool.execute(ShuiRunnableImpl(this))
                }
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        println(">>>>>>>>>>服务被销毁了..")
        ShuiThreadPool.shutdown()
    }

}